src/transaction.rs

```diff
@@ -106,6 +107,39 @@ impl CommitTransaction {
     pub fn value(&self) -> Amount {
         Amount::from_sat(self.0.output[0].value)
     }
+
+    fn descriptor(
+        (X_0, R_0, Y_0): (PublicKey, RevocationPublicKey, PublishingPublicKey),
+        (X_1, R_1, Y_1): (PublicKey, RevocationPublicKey, PublishingPublicKey),
+        time_lock: u32,
+    ) -> Result<miniscript::Descriptor<bitcoin::PublicKey>> {
+        let X_0 = hex::encode(X_0.serialize().to_vec());
+        let R_0 = hex::encode(Into::<PublicKey>::into(R_0).serialize().to_vec());
+        let Y_0 = hex::encode(Into::<PublicKey>::into(Y_0).serialize().to_vec());
+        let X_1 = hex::encode(X_1.serialize().to_vec());
+        let R_1 = hex::encode(Into::<PublicKey>::into(R_1).serialize().to_vec());
+        let Y_1 = hex::encode(Into::<PublicKey>::into(Y_1).serialize().to_vec());
+
+        // Describes the spending policy of the channel commit transaction T_c.
+        // There are possible way to spend this transaction:
+        // 1. Channel state: It is correctly signed w.r.t pk_0, pk_1 and after relative timelock
+        // 2. Punish 0: It is correctly signed w.r.t pk_1, Y_0, R_0
+        // 3. Punish 1: It is correctly signed w.r.t pk_0, Y_1, R_1
+        // TODO: Make an explicit choice between using `tresh` or composing `or` and `and` (which are both binary operators)
```

Comment:
For the funding transaction I went for the `AND` approach.

I would also go for `AND/OR` here because according to the docs (see below) all policies in `thresh` are equally likely while I would assume that the happy path (collaboratively signed) is the one we should optimize for. Either way, this is just an optimization and we don't have to worry about this for now. 

----

http://bitcoin.sipa.be/miniscript/: 
`thresh(NUM,POL,POL,...): Require that NUM out of the following subpolicies are met (all combinations are assumed to be equally likely).`

Reply:
I am happy to update it.

Reply:
The script changes between runs when using `or` and `and`. Shouldn't it be stable?